app/src/main/java/com/example/splendormobilegame/activities/GameActivity/DeckReservingController.java

```diff
@@ -44,11 +63,25 @@ public class ReservationFromDeckMessageHandler extends UserReaction {
         public UserMessage react(ServerMessage serverMessage) {
             Log.i("UserReaction", "Entered ReservationFromDeckMessageHandler react method");
 
-            // TODO Update the model
+            
+            MakeReservationFromDeck.ResponseData responseData = (MakeReservationFromDeck.ResponseData) ReactionUtils.getResponseData(serverMessage, MakeReservationFromDeck.ResponseData.class);
+
+
+            User user=Model.getRoom().getUserByUuid(Model.getUserUuid());
+            Card card=Model.getRoom().getGame().getCardByUuid(responseData.card.uuid);
```

Comment:
Card with this `uuid` does not exist in our `Model` on android. This card is a brand new card received from server. Server response has all needed data about this card to create its object on android and put it into `Model`. So please, use these pieces of information and create new `Card` object.